components/cluster/command/deploy.go

```diff
@@ -146,6 +146,12 @@ func confirmTopology(clusterName, version string, topo *spec.Specification, patc
 		log.Errorf("    3. The component marked as `patched` has been replaced by previours patch command.")
 	}
 
+	if (topo.TiSparkMasters != nil && len(topo.TiSparkMasters) > 0) ||
```

Comment:
This part can be simplified to `if len(topo.TiSparkMasters) > 0) || len(topo. TiSparkWorkers) > 0)` because `len(nil)` always returning 0.